test/test_find_library.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+
+#include "rcpputils/find_library.hpp"
+
+namespace rcpputils
+{
+namespace
+{
+
+TEST(test_find_library, find_library)
+{
+  // Get ground-truth values from CTest properties.
+  const char * toy_lib_expected = getenv("_TOY_TEST_LIBRARY");
+  EXPECT_NE(toy_lib_expected, nullptr);
+  const char * toy_lib_dir = getenv("_TOY_TEST_LIBRARY_DIR");
+  EXPECT_NE(toy_lib_dir, nullptr);
+
+  // Set our relevant path variable.
+  const char * env_var{};
+#ifdef _WIN32
```

Comment:
If you modify the code as follow, this file code will be completely evaluated by the compiler on any platform which would help us catching errors more quickly.

```c++
#ifdef WIN32
static constexpr bool is_win32 = true;
#else
static constexpr bool is_win32 = false;
#endif
[...]

if (is_win32) {
  env_var = ...
} else {
  env_var = ..
}
```

Reply:
Hoisting to top-level removed the need for this.